### PR TITLE
Feature/map

### PR DESCRIPTION
**Pull request by zombiechook**
- Minor bugfixes with the graphics of the maps
 - Implemented sliding on ice tiles for the ice map
 - Added secret area for ice map
 - Created fire map with basic secret area
 - Implemented damage over time in certain spots on a map
 - Implemented making the player move slower over time when in a "freezing area"

### CODE REVIEW

**Review by AnshulDadhwal — APPROVED** [review]
Looks good to me 👍

File: source/core/assets/levels/hell/main-area/entities/treeTop.json
Comment (Foref) [inline source/core/assets/levels/hell/main-area/entities/treeTop.json:1]: Empty file?

File: source/desktop/out/production/resources/saves/savefile.json
Comment (Foref) [inline source/desktop/out/production/resources/saves/savefile.json:1]: Possible index error here

File: source/core/assets/levels/hell/main-area/entities/ship.json
Comment (Foref) [inline source/core/assets/levels/hell/main-area/entities/ship.json:1]: Check json files if they are missing the string ""

Comment (Foref) [reply source/core/assets/levels/hell/main-area/entities/ship.json:1]: example,

 "class": "..."

File: source/core/assets/levels/hell/main-area/entities/treeTop.json
Comment (zombiechook) [reply source/core/assets/levels/hell/main-area/entities/treeTop.json:1]: Intentional. File can safely be deleted

File: source/desktop/out/production/resources/levels/earth/main-area/entities/astro.json
Comment (Foref) [inline source/desktop/out/production/resources/levels/earth/main-area/entities/astro.json:1]: json missing string

File: source/desktop/out/production/resources/levels/frozen/main-area/main.json
Comment (Foref) [inline source/desktop/out/production/resources/levels/frozen/main-area/main.json:1]: is box_boy and the rest of the original forestgameArea assets used?

File: saves/savefile.json
Comment (Foref) [inline saves/savefile.json:1]: Indentation error

File: source/desktop/out/production/resources/levels/earth/main-area/entities/spawners.json
Comment (kk4w4i) [inline source/desktop/out/production/resources/levels/earth/main-area/entities/spawners.json:1]: For spawner configs the "sound" variable needs to be "sounds" just the syntax with our configs :)

File: source/desktop/out/production/resources/levels/frozen/main-area/entities/spawners.json
Comment (kk4w4i) [inline source/desktop/out/production/resources/levels/frozen/main-area/entities/spawners.json:1]: Same for this :)

File: source/desktop/out/production/resources/levels/lush/main-area/entities/spawners.json
Comment (kk4w4i) [inline source/desktop/out/production/resources/levels/lush/main-area/entities/spawners.json:1]: And this :)

Comment (rike568) [reply source/desktop/out/production/resources/levels/lush/main-area/entities/spawners.json:1]: Just added the s, please have a look again, thanks

File: source/desktop/out/production/resources/levels/earth/main-area/entities/astro.json
Comment (rike568) [reply source/desktop/out/production/resources/levels/earth/main-area/entities/astro.json:1]: I had a look and are .json files are the exact same as main as they don't also have the strings. Please have a look again, thanks

File: source/desktop/out/production/resources/levels/lush/main-area/entities/spawners.json
Comment (kk4w4i) [reply source/desktop/out/production/resources/levels/lush/main-area/entities/spawners.json:1]: lgtm :)

**Review by kk4w4i — APPROVED** [review]

**Review by OjaswinKohli — APPROVED** [review]
Really good work. I think all files are okay.

**Pull request merged**